### PR TITLE
Feature/favourite recipes feature upgrade

### PR DESCRIPTION
**Pull request by martynaspuskunigis**

### CODE REVIEW

File: src/app/active-recipe-container.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            }
+            case Abstractions.ItemStatus.NoData: {
+                return <div>No data.</div>;
+            }
+            case Abstractions.ItemStatus.Failed: {
+                return (
+                    <div>
+                        Failed to load...
+                        <span>
+                            <button>Retry...</button>
```
Comment (DeividasBakanas) [inline src/app/active-recipe-container.tsx]: Is this fixed in other interface?

File: src/app/components/fav-recipe-item-view.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import * as React from "react";
+import { Link } from "react-router-dom";
+
+import { RecipesActionsCreators } from "./../recipes-actions-creators";
```
Comment (DeividasBakanas) [inline src/app/components/fav-recipe-item-view.tsx]: Simplify paths.

File: src/app/components/fav-recipe-item-view.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import { Recipe } from "./../contracts/Recipe";
```
Comment (DeividasBakanas) [inline src/app/components/fav-recipe-item-view.tsx]: Simplify paths.

File: src/app/recipes-store.ts [hunk 4]
```diff
@@ -55,8 +50,7 @@ class RecipesReduceStoreClass extends ReduceStore<StoreState> {
         return {
             recipes: [],
             status: Abstractions.ItemStatus.Init,
-            activeRecipe: "",
-            favoriteRecipes: []
+            activeRecipe: ""
```
Comment (DeividasBakanas) [inline src/app/recipes-store.ts]: `activeRecipeId`? Use `undefined` for initial value.

File: src/app/recipes-item-container.tsx [hunk 5]
```diff
@@ -34,20 +34,23 @@ class RecipesItemContainerClass extends React.Component<Props, State> {
...
             case Abstractions.ItemStatus.NoData: {
                 return <div>No data.</div>;
             }
             case Abstractions.ItemStatus.Failed: {
                 return (
                     <div>
-                        Failed to load... <span><button>Retry...</button></span>
+                        Failed to load...
+                        <span>
+                            <button>Retry...</button>
```
Comment (DeividasBakanas) [inline src/app/recipes-item-container.tsx]: Is this fixed in other pull request?

File: src/app/fav-recipes-store.ts [hunk 6]
```diff
@@ -27,26 +24,36 @@ class FavRecipesReduceStoreClass extends ReduceStore<StoreState> {
...
             ...state
         };
         nextState.favRecipes.push(action.recipeId);
         return {
-            ...state,
             favRecipes: [...nextState.favRecipes]
         };
     }
 
+    public getLocalStorage(): string[] {
```
Comment (DeividasBakanas) [inline src/app/fav-recipes-store.ts]: Improve naming. Local storage can be get using `localStorage` or `window.localStorage`. Your function have different behavior ;)

File: src/app/fav-recipes-store.ts [hunk 7]
```diff
@@ -27,26 +24,36 @@ class FavRecipesReduceStoreClass extends ReduceStore<StoreState> {
...
+        const favRecipes: string[] = [];
+        for (let i = 0; i < localStorage.length; i++) {
+            const key = localStorage.key(i);
+            if (key != null && localStorage.getItem(key) != null) {
+                const item = localStorage.getItem(key)!;
```
Comment (DeividasBakanas) [inline src/app/fav-recipes-store.ts]: const item = localStorage.getItem(key)**!**;

Why using `!`? Remove or add a comment.

File: src/app/fav-recipes-item-container.tsx [hunk 8]
```diff
@@ -2,51 +2,55 @@ import * as React from "react";
...
             case Abstractions.ItemStatus.NoData: {
                 return <div>No data.</div>;
             }
             case Abstractions.ItemStatus.Failed: {
                 return (
                     <div>
-                        Failed to load... <span><button>Retry...</button></span>
+                        Failed to load...
+                        <span>
+                            <button>Retry...</button>
```
Comment (DeividasBakanas) [inline src/app/fav-recipes-item-container.tsx]: Is this implemented in other pull request?

File: src/app/fav-recipes-container.tsx [hunk 9]
```diff
@@ -17,22 +14,15 @@ class RecipesContainerClass extends React.Component<{}, State> {
...
 
     public render(): JSX.Element | JSX.Element[] {
-        if (this.state.status === Abstractions.ItemStatus.Pending) {
-            return <Spinner/>;
-        }
-        if (this.state.recipes == null || this.state.recipes.length === 0) {
-            return <div>No results found...</div>;
-        }
-        const recipeList = this.state.recipes.map((recipeId, index) => (
+        const recipeList = this.state.recipesIds.map((recipeId, index) => (
```
Comment (DeividasBakanas) [inline src/app/fav-recipes-container.tsx]: Inline `recipeList`.

File: src/app/components/home-view.tsx [hunk 10]
```diff
@@ -1,15 +1,14 @@
 import * as React from "react";
 
-import { SearchView } from "./search-view";
 import { RecipesContainer } from "../recipes-container";
```
Comment (DeividasBakanas) [inline src/app/components/home-view.tsx]: Imports order.

File: src/app/components/fav-recipe-item-view.tsx [hunk 11]
```diff
@@ -1,45 +1,35 @@
...
         return (
             <div className="recipes__box">
-                <img className="recipe__box-img" src={this.props.recipe.image_url} alt={this.props.recipe.title} />
+                <img className="recipe__box-img" src={this.props.recipeToDisplay.image_url} alt={this.props.recipeToDisplay.title} />
                 <div className="recipe__text">
                     <h5 className="recipes__title">
-                        {this.props.recipe.title.length < 20
-                            ? `${this.props.recipe.title}`
-                            : `${this.props.recipe.title.substring(0, 25)}...`}
+                        {this.props.recipeToDisplay.title.length < 20
```
Comment (DeividasBakanas) [inline src/app/components/fav-recipe-item-view.tsx]: Ellipsis should be implemented from styles side.

File: src/app/components/fav-recipe-item-view.tsx [hunk 12]
```diff
@@ -1,45 +1,35 @@
 import * as React from "react";
 import { Link } from "react-router-dom";
 
-import { Recipe } from "./../contracts/Recipe";
 import { RecipesActionsCreators } from "./../recipes-actions-creators";
+import { Recipe } from "./../contracts/Recipe";
```
Comment (DeividasBakanas) [inline src/app/components/fav-recipe-item-view.tsx]: Simplify path.

File: src/app/components/fav-recipe-item-view.tsx [hunk 13]
```diff
@@ -1,45 +1,35 @@
 import * as React from "react";
 import { Link } from "react-router-dom";
 
-import { Recipe } from "./../contracts/Recipe";
 import { RecipesActionsCreators } from "./../recipes-actions-creators";
```
Comment (DeividasBakanas) [inline src/app/components/fav-recipe-item-view.tsx]: Simplify path.

File: src/app/active-recipe-container.tsx [hunk 14]
```diff
@@ -46,7 +46,7 @@ class ActiveRecipeContainerClass extends React.Component<Props, State> {
             case Abstractions.ItemStatus.Failed: {
                 return (
                     <div>
-                        Failed to load...{" "}
+                        Failed to load...
                         <span>
                             <button>Retry...</button>
```
Comment (DeividasBakanas) [inline src/app/active-recipe-container.tsx]: Is this implemented in other pull request?

File: src/app/active-recipe-container.tsx (hunk 1 above)
Comment (martynaspuskunigis) [reply src/app/active-recipe-container.tsx]: This is fixed in another pull request.

File: src/app/recipes-item-container.tsx (hunk 5 above)
Comment (martynaspuskunigis) [reply src/app/recipes-item-container.tsx]: This is fixed in another pull request.

File: src/app/fav-recipes-item-container.tsx (hunk 8 above)
Comment (martynaspuskunigis) [reply src/app/fav-recipes-item-container.tsx]: Yes :)

**Pull request merged**